### ISSUE TITLE
Duda clean code

### ISSUE THREAD

**Issue by CataAlamos** [RodrigoToroIcarte/IIC2113-2023-1#70]
Hola! No me queda claro cuando es conveniente usar bools vs try and catch. Por ejemplo, 

bool isReversal = reversal(card) 

if(isReversal) 
{
hace algo 
}

vs 
try {
reversal(card) 
hace algo
}
catch {
}

**Comment by RodrigoToroIcarte** [comment]
Si el catch está vacío probablemente bastaba con un if. Usa excepciones en vez de retornar un código de error. Un código de error es cuando un método hace algo y además retorna otra cosa para indicar si lo que debía hacer fue exitoso o no. Ese "algo que se retorna" podría o no ser un booleano. En situaciones como esas lo mejor es usar try-catch. 

... pero en el ejemplo que muestras probablemente basta con un if.

**Comment by CataAlamos** [comment]
Pero ese catch no podría ser un else? Porque al final el if indica si se logro lo esperado, y en caso contrario se debe hacer otra cosa.

**Comment by RodrigoToroIcarte** [comment]
Claro. Los ifs y else son buenos. Prefiere eso a un try-catch de ser posible. Lo que ocurre es que si esos if-else se deben a que estás manejando códigos de error entonces es mejor usar un try-catch. Pero igual me cuesta un poco tener esta discusión sin un ejemplo concreto de código. En el ejemplo que das sobre los reversals, eso pareciera ser un caso donde un if-else es lo apropiado.

De hecho, pensando en la E3, para mí el único caso donde me hace sentido usar un try-catch es para lo del mazo inválido. En otro lugar lo siento un poco forzado (aunque eso no significa que esté necesariamente malo su uso en otra situación de la E3).

**Comment by CataAlamos** [comment]
Estaba pensando, para cartas reversal. Tipo si se pudo revertir la carta se hace algo sino no. En ese caso, es un bool porque no estamos atrapando un error sino que es una condición del juego?

**Comment by RodrigoToroIcarte** [comment]
Depende de cómo lo tienes modelado. Yo modelé eso usando un if no más. Pero puedo imaginar que si asumes que toda carta es exitosa pero tiras un error cuando es revertida, quizás el código quede más limpio. Así que lo dejo a tu criterio. Lo único es que no quiero que abusen de lanzar excepciones por cualquier cosa :P. Que lanzar una excepción sea un último recurso para limpiar un código que está quedando enredado debido al constante manejo de casos de error vía if-else.
